autogpt/workspace.py

```diff
@@ -39,7 +39,11 @@ def safe_path_join(base: Path, *paths: str | Path) -> Path:
     """
     joined_path = base.joinpath(*paths).resolve()
 
-    if CFG.restrict_to_workspace and not joined_path.is_relative_to(base):
+    is_not_joined_path_relative = (
+        callable(getattr(Path, "is_relative_to", None))
+        and not joined_path.is_relative_to(base)
+    ) or not joined_path.relative_to(base)
```

Comment:
`path_a.relative_to(path_b)` is not a boolean statement.  It makes a new path in the form `path_a / path_b`.  This does not do quite what you want, therefore.